pkg/build/controller/controller.go

```diff
@@ -85,20 +87,16 @@ func (bc *BuildController) nextBuildStatus(build *buildapi.Build) error {
 			namespace = build.Namespace
 		}
 
-		repo, err := bc.ImageStreamClient.GetImageStream(namespace, ref.Name)
+		var err error
+		spec, err = bc.resolveImageStreamReference(ref, namespace)
```

Comment:
is there a good reason we didn't resolve all this when we created the build object, instead of waiting until now to do it?